netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -585,6 +585,7 @@ private void sendGrpcFrame(ChannelHandlerContext ctx, SendGrpcFrameCommand cmd,
     }
     // Call the base class to write the HTTP/2 DATA frame.
     encoder().writeData(ctx, cmd.streamId(), cmd.content(), 0, cmd.endStream(), promise);
+    transportTracer.reportMessageSent(cmd.getNumMessages());
```

Comment:
It is not guaranteed numMessages will be positive here. For large messages it will be zero for many frames. I'd suggest requiring non-negative instead. It looks like the previous code was simply broken.

Reply:
Changed to checking for `>= 0`